### PR TITLE
SUS-1675 | allow 5000 results on Special:ListRedirects and Special:Mostlinked

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-1675

Now Both `Mostlinked` and `Listredirects` special pages return a larger set of results.

### CODE REVIEW

File: includes/QueryPage.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
  */
 global $wgQueryPages; // not redundant
```
Comment (WikiaTech) [inline includes/QueryPage.php:15]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 5 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [QueryPage.php#L55](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L55): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [QueryPage.php#L65](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L65): Class "QueryPage" has 31 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [QueryPage.php#L329](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L329): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [QueryPage.php#L449](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L449): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [QueryPage.php#L747](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L747): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![INFO][INFO] [QueryPage.php#L315](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L315): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [QueryPage.php#L474](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L474): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. ![INFO][INFO] [QueryPage.php#L537](https://github.com/Wikia/app/blob/dcb02f7e931e5ea694940e2116bc4ff6ec1f0f6e/includes/QueryPage.php#L537): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 5 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [QueryPage.php#L55](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L55): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [QueryPage.php#L65](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L65): Class "QueryPage" has 31 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [QueryPage.php#L329](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L329): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [QueryPage.php#L449](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L449): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [QueryPage.php#L747](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L747): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![INFO][INFO] [QueryPage.php#L315](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L315): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [QueryPage.php#L474](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L474): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. ![INFO][INFO] [QueryPage.php#L537](https://github.com/Wikia/app/blob/c472ef5bf93011bfa8cf276dd4e1ac16db4dbaee/includes/QueryPage.php#L537): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: includes/QueryPage.php [hunk 2]
```diff
@@ -443,11 +443,44 @@ public function getCachedTimestamp() {
...
+		}
+
+		return $limit;
+	}
+
+	/**
 	 * This is the actual workhorse. It does everything needed to make a
 	 * real, honest-to-gosh query page.
 	 */
 	function execute( $par ) {
```
Comment (WikiaTech) [inline includes/QueryPage.php:482]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 5 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [QueryPage.php#L55](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L55): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [QueryPage.php#L65](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L65): Class "QueryPage" has 32 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [QueryPage.php#L329](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L329): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [QueryPage.php#L780](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L780): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![INFO][INFO] [QueryPage.php#L315](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L315): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [QueryPage.php#L507](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L507): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. ![INFO][INFO] [QueryPage.php#L570](https://github.com/Wikia/app/blob/fefa7c9894c2c34e216577e7f66def938b245833/includes/QueryPage.php#L570): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by mszabo-wikia — APPROVED** [review]

File: includes/QueryPage.php [hunk 3]
```diff
@@ -443,11 +443,44 @@ public function getCachedTimestamp() {
...
+	 */
+	protected function getMaxResults() {
+		global $wgQueryCacheLimit, $wgQueryPages;
+
+		// that's our default return value
+		$limit = $wgQueryCacheLimit;
+
+		// get the $wgQueryPages entry for the current QueryPage class
+		$currentQueryPage = get_class( $this );
+		$entries = array_values( array_filter(
```
Comment (pwojcik86) [inline includes/QueryPage.php:463]: Does `array_values` do anything here?

Comment (macbre) [reply includes/QueryPage.php:463]: `array_filter` keeps the key from the original array. `array_values` will make sure that we can easily use `[0]` to get the only item in `$entries`.

**Pull request merged**